clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1235,9 +1236,15 @@ public void assign(Collection<TopicPartition> partitions) {
                     }
                 }
 
-                final Map<TopicPartition, List<ConsumerRecord<K, V>>> records = pollForFetches(timer);
-                if (!records.isEmpty()) {
-                    // before returning the fetched records, we can send off the next round of fetches
+                final Fetch<K, V> fetch = pollForFetches(timer);
+                if (!fetch.isEmpty()) {
+                    if (fetch.records().isEmpty()) {
+                        log.debug(
```

Comment:
I'm not so sure about the value of this log line. Maybe it would be more useful to ensure that we have enough logging in `Fetcher` that we can see when the position gets advanced without any record data? That would tell us not only why a poll() returned with empty data, but which partition caused it to do so. What do you think?

Reply:
I think that works. My goal with this line was to help people understand the cause for this new behavior, as it may seem like a bug at first if `poll` returns empty batches before the poll timeout has elapsed. I've taken a stab at a message that has functional value (i.e., naming the specific topic partition whose position has advanced without any user-visible records) and still tries to let the user know about how/why this may affect `poll` behavior; LMKWYT.

Also, this section was pretty buggy as-was since it skipped the call to `client.transmitSends` and didn't pass the empty batch through the consumer's interceptors. I've fixed both of these; LMK if you think we should skip passing an empty batch to interceptors, though.